.pre-commit-config.yaml

```diff
@@ -40,5 +40,13 @@ repos:
       - id: poetry-check
         args: ['-C', 'api']
 
+  - repo: local
+    hooks:
+      - id: mypy-check
+        name: mypy check
+        entry: bash -c "cd api && poetry run python scripts/mypy_check.py"
```

Comment:
@zachaysan I'm not sure the latest changes make any reference to this?

Reply:
I've altered the script to run the files one at a time. I had to use `git diff` to list the target files of the commit because no matter what I tried to pass into the bash script it just wouldn't run with the automatic listed files. The good news is that we are now doing a `mypy` check on only the committed files.